tests/foreman/cli/test_host.py

```diff
@@ -2076,6 +2078,188 @@ def test_negative_multi_cv_registration(
+    host = session_multicv_sat.cli.Host.info({'name': rhel_contenthost.hostname})
+    assert (
+        len(host['content-information']['content-view-environments']) == 2
+    ), "Expected host to be registered to both environments"
+
+
+@pytest.mark.rhel_ver_match('[^7]')
+def test_positive_multi_cv_host_repo_availability(
+    session_multicv_sat,
+    rhel_contenthost,
+):
+    """Multi-environment hosts should have access to repositories from all of their content view environments.
+
+    :id: 6a15d591-be84-4b5b-8deb-8ea6eb32fee6
+
+    :steps:
+        1. Create two synced custom repositories
+        2. Add each repository to its own content view. Publish and promote the content views.
+        3. Register a host with global registration to a single content view environment
+        4. Expose the allow_multiple_content_views setting
+        5. Set the allow_multiple_content_views setting to true
+        6. Assign the host to multiple content view environments
+        7. Confirm repos listed in subscription-manager repos match the repos in the content view environments
+
+    :expectedresults: Host sees repositories from both content view environments, not just one.
+
+    :CaseImportance: Critical
+
+    :parametrized: yes
+    """
+
+    # Create an org
+    multicv_org = session_multicv_sat.api.Organization().create()
+
+    # Create activation key
+    multicv_ak = session_multicv_sat.api.ActivationKey(
+        content_view=multicv_org.default_content_view.id,
+        environment=multicv_org.library.id,
+        organization=multicv_org,
+    ).create()
+
+    # Create a lifecycle environment
+    multicv_lce = session_multicv_sat.api.LifecycleEnvironment(organization=multicv_org).create()
+
+    # Create a content view
+    cv1 = session_multicv_sat.api.ContentView(organization=multicv_org).create()
+    cv1.publish()
+    cv1 = cv1.read()
+
+    # Create two synced custom repositories
+    repo_instances = []
+    for repo in ['RepoA', 'RepoB']:
+        product_name = f'{repo}_Product_{str(uuid.uuid4())}'
+        repo_id = session_multicv_sat.api_factory.create_sync_custom_repo(
+            org_id=multicv_org.id,
+            product_name=product_name,
+            repo_name=repo,
+            repo_url=settings.repos.fake_repo_zoo3,
+        )
+        repo_instances.append(session_multicv_sat.api.Repository(id=repo_id).read())
+
+    repo_b = repo_instances.pop()
+    repo_a = repo_instances.pop()
+
+    # Add one repository each to two content views. Publish and promote the content views.
+    cv1.repository = [repo_a]
+    cv1 = cv1.update(['repository'])
+    cv1.publish()
+    # for some reason I have to call read() or it doesn't pick up the new version
+    module_published_cvv = cv1.read().version[0]
+    module_published_cvv.promote(data={'environment_ids': multicv_lce.id})
+
+    lce2 = session_multicv_sat.api.LifecycleEnvironment(organization=multicv_org).create()
```

Comment:
I got lost a bit, does the `lce2` base on `Library` or `multicv_lce`. If on `Library` (as I believe), could we rename `multicv_lce` to `lce1`?